test/test_image.py

```diff
@@ -0,0 +1,261 @@
+        input_ = io.BytesIO(open("test/images/alpha.png", "rb").read())
+        output = io.BytesIO()
+        yoga.image.optimize(input_, output)
+        output.seek(0)
+        assert output.read().startswith(_MAGIC_PNG)
+
+    def test_output_file(self, tmpdir):
+        # str (path)
+        output_path = os.path.join(str(tmpdir), "output1.png")
+        yoga.image.optimize("test/images/alpha.png", output_path)
+        output = open(output_path, "rb")
+        assert output.read().startswith(_MAGIC_PNG)
+
+        # file
+        output_path = os.path.join(str(tmpdir), "output2.png")
+        output = open(output_path, "wb")
+        yoga.image.optimize("test/images/alpha.png", output)
+        output.close()
+        output = open(output_path, "rb")
+        assert output.read().startswith(_MAGIC_PNG)
+
+        # ByteIO
+        output = io.BytesIO()
+        yoga.image.optimize("test/images/alpha.png", output)
+        output.seek(0)
+        assert output.read().startswith(_MAGIC_PNG)
+
+    def test_option_output_format_default(self):
+        # JPEG
+        output = io.BytesIO()
+        yoga.image.optimize("test/images/image1.jpg", output)
+        output.seek(0)
+        assert output.read().startswith(_MAGIC_JPEG)
+
+        # PNG
+        output = io.BytesIO()
+        yoga.image.optimize("test/images/unused-alpha.png", output)
+        output.seek(0)
+        assert output.read().startswith(_MAGIC_PNG)
+
+    def test_option_output_format_orig(self):
+        # JPEG
+        output = io.BytesIO()
+        yoga.image.optimize("test/images/image1.jpg", output, {
+            "output_format": "orig"
+            })
+        output.seek(0)
+        assert output.read().startswith(_MAGIC_JPEG)
+
+        # PNG
+        output = io.BytesIO()
+        yoga.image.optimize("test/images/unused-alpha.png", output, {
+            "output_format": "orig"
+            })
+        output.seek(0)
+        assert output.read().startswith(_MAGIC_PNG)
+
+        # Other
+        output = io.BytesIO()
+        with pytest.raises(ValueError):
+            yoga.image.optimize("test/images/image.gif", output, {
+                "output_format": "orig"
+                })
+
+    @pytest.mark.skip(reason="will be implemented later")
+    def test_option_output_format_auto(self):
+        raise NotImplementedError()  # TODO
+        # jpeg -> jpeg
+        # png -> jpeg
+        # png -> png
+        # jpeg -> png
+
+    def test_option_output_format_jpeg(self):
+        # JPEG
+        output = io.BytesIO()
+        yoga.image.optimize("test/images/image1.jpg", output, {
+            "output_format": "jpeg"
+            })
+        output.seek(0)
+        assert output.read().startswith(_MAGIC_JPEG)
+
+        # PNG
+        output = io.BytesIO()
+        yoga.image.optimize("test/images/unused-alpha.png", output, {
+            "output_format": "jpeg"
+            })
+        output.seek(0)
+        assert output.read().startswith(_MAGIC_JPEG)
```

Comment:
Which behavior if it's a PNG with a useful alpha channel in input but JPG output is asked?

Reply:
The image will be overlaid on a black background and converted to jpeg (it is implemented in pyguetzli → https://github.com/wanadev/pyguetzli/blob/master/pyguetzli/pil_image.py#L13 )